### PR TITLE
[FLUX] independently drop t5 and clip tokens for classifier free guidance 

### PR DESCRIPTION
**Pull request by wwwjn**
In the SD3 paper, independently drop each tokenzier's result for classifier free guidance in trainnig.
> For unconditional diffusion guidance (Ho & Salimans, 2022), we set the outputs of each of the three text encoders independently to zero with a probability of 46.4%, such that we roughly train an unconditional model in 10% of all steps.

Thanks @tianyu-l  for identifying this issue!

### CODE REVIEW

**Review by tianyu-l — APPROVED** [review]

**Review by tianyu-l — COMMENTED** [review]
I think we also want to understand the default dropout probability.
Currently in schnell it's 10%, but to make it marginally 10% we need something else.
We need X so that X^2 = 10%?

**Comment by wwwjn** [comment]
> I think we also want to understand the default dropout probability. Currently in schnell it's 10%, but to make it marginally 10% we need something else. We need X so that X^2 = 10%?

We need to change the parameter to 0.316

File: torchtitan/experiments/flux/train_configs/debug_model.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ max_norm = 2.0  # grad norm clipping
 steps = 10
 compile = false
 dataset = "cc12m-test"
-classifer_free_guidance_prob = 0.1
+classifer_free_guidance_prob = 0.316
```
Comment (tianyu-l) [inline torchtitan/experiments/flux/train_configs/debug_model.toml]: let's add a comment to explain the reason of this choice, maybe in JobConfig where this is defined.

Shall we also modify
https://github.com/pytorch/torchtitan/blob/b6da6ecc7560c9b432b93a1f13e1033aefcbcc58/torchtitan/experiments/flux/tests/test_generate_image.py#L49
https://github.com/pytorch/torchtitan/blob/b6da6ecc7560c9b432b93a1f13e1033aefcbcc58/torchtitan/experiments/flux/tests/unit_tests/test_flux_dataloader.py#L47

Also are we using 0.316 universally? If so can we just use it as default, instead of 0?

**Review by tianyu-l — CHANGES_REQUESTED** [review]

File: torchtitan/experiments/flux/train_configs/debug_model.toml (hunk 1 above)
Comment (wwwjn) [reply torchtitan/experiments/flux/train_configs/debug_model.toml]: I would prefer still using 0 as default value, this also aligns with `eval` parameters (during inference, the default value is "enable_classifer_free_guidance = false"). Users should explicitly aware they enabled classifier-free guidance
